pkg/units/control.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package units implements the business logic to make the fuzzer work
+package units
+
+import (
+	"fmt"
+
+	//"buzzer/pkg/strategies/strategies"
+	"buzzer/pkg/ebpf/ebpf"
+	epb "buzzer/proto/ebpf_go_proto"
+	fpb "buzzer/proto/ffi_go_proto"
+)
+
+// StrategyInterface contains all the methods that a fuzzing strategy should
+// implement.
+type StrategyInterface interface {
+	// GenerateProgram should return the instructions to feed the verifier.
+	GenerateProgram(ffi *FFI) ([]*epb.Instruction, error)
```

Comment:
Does GenerateProgram need an ffi? It's not clear to me what this is used for.

Reply:
Yes, sometimes programs need to interact with maps (e.g create them and embed the fd in the bytecode). In fact I think all methods need a reference to the FFI in case they need to interact with ebpf via C